app/src/main/java/com/example/bigowlapp/utils/AuthenticatorByPhoneNumber.java

```diff
@@ -0,0 +1,66 @@
+package com.example.bigowlapp.utils;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.telephony.TelephonyManager;
+import android.widget.Toast;
+
+import com.example.bigowlapp.model.Attendance;
+import com.example.bigowlapp.model.AuthByPhoneNumberFailure;
+import com.example.bigowlapp.model.Schedule;
+import com.example.bigowlapp.model.User;
+import com.example.bigowlapp.model.UserScheduleResponse;
+import com.example.bigowlapp.repository.AuthRepository;
+import com.example.bigowlapp.repository.RepositoryFacade;
+import com.google.firebase.Timestamp;
+
+public class AuthenticatorByPhoneNumber {
+
+    private String deviceIdNumber;
+    private String devicePhoneNumber;
+    private String currentUserPhoneNumber;
+    private final Context context;
+    private final AuthRepository authRepository = new AuthRepository();
+
+    RepositoryFacade repositoryFacade = RepositoryFacade.getInstance();
+
+    public AuthenticatorByPhoneNumber(Context context) {
+        this.context = context;
+    }
+
+    @SuppressLint("MissingPermission")
+    public void authenticate(String scheduleId) {
+        TelephonyManager telephonyManager = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
+        deviceIdNumber = telephonyManager.getDeviceId();
+        devicePhoneNumber = telephonyManager.getLine1Number();
```

Comment:
there was a READ_SMS permission seems to be necessary for `getLine1Number()`, so you'll have to add that permission as well

Reply:
It seems to be necessary for SDK API level 30 and above, so I have added it

Reply:
Seems like @SuppressLint({"MissingPermission", "HardwareIds"}) is needed on line31 instead